contracts/pegged-bridge/legacy/NativeTokenVaultProxy.sol

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity 0.8.9;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "../../libraries/PbPegged.sol";
+import "../../interfaces/IOriginalTokenVault.sol";
+import "../../interfaces/IWETH.sol";
+
+// To support single-transaction deposit of native gas token (e.g., ETH in ethereum)
+// to legacy vaults that do not have the depositNative function
+// Note that the depositor address shown at the desitnation chain will be this contract address
+contract NativeTokenVaultProxy {
+    address public immutable nativeWrap;
+    address public immutable vault;
+
+    mapping(bytes32 => address) public senders;
+
+    constructor(address _nativeWrap, address _vault) {
+        nativeWrap = _nativeWrap;
+        vault = _vault;
+    }
+
+    /**
+     * @notice Deposit native token in the vault
+     * @param _amount The amount to deposit.
+     * @param _mintChainId The destination chain ID to mint tokens.
+     * @param _mintAccount The destination account to receive the minted pegged tokens.
+     * @param _nonce A number input to guarantee unique depositId. Can be timestamp in practice.
+     */
+    function depositNative(
+        uint256 _amount,
+        uint64 _mintChainId,
+        address _mintAccount,
+        uint64 _nonce
+    ) external payable returns (bytes32) {
+        require(msg.value == _amount, "Amount mismatch");
+        require(nativeWrap != address(0), "Native wrap not set");
+        bytes32 depositId = keccak256(
+            abi.encodePacked(
+                address(this),
+                nativeWrap,
+                _amount,
+                _mintChainId,
+                _mintAccount,
+                _nonce,
+                uint64(block.chainid)
+            )
+        );
+        IWETH(nativeWrap).deposit{value: _amount}();
+        IOriginalTokenVault(vault).deposit(nativeWrap, _amount, _mintChainId, _mintAccount, _nonce);
+        senders[depositId] = msg.sender;
```

Comment:
Emit an event to indicate the "real" depositor?

Reply:
not really useful? Especially given that the real depositor is recorded on chain

Reply:
For history monitoring?

Reply:
added